source/common/common/utility.cc

```diff
@@ -322,18 +322,25 @@ absl::string_view StringUtil::cropLeft(absl::string_view source, absl::string_vi
 
 std::vector<absl::string_view> StringUtil::splitToken(absl::string_view source,
                                                       absl::string_view delimiters,
-                                                      bool keep_empty_string) {
+                                                      bool keep_empty_string,
+                                                      bool trim_whitespace) {
+  std::vector<absl::string_view> result;
   if (keep_empty_string) {
-    return absl::StrSplit(source, absl::ByAnyChar(delimiters));
+    result = absl::StrSplit(source, absl::ByAnyChar(delimiters));
+  } else {
+    result = absl::StrSplit(source, absl::ByAnyChar(delimiters), absl::SkipEmpty());
+  }
+
+  if (trim_whitespace) {
+    for_each(result.begin(), result.end(), [](auto& v) { v = trim(v); });
```

Comment:
So if we have
"a, ,b"
and trim whitespace is true and keep empty string is false, we will end up returning an empty string, no?
Mind fixing and adding a unit test?

Reply:
Done! Thanks for catching the bug! I added a test for this case.